Shared/Samples/Identify graphics/IdentifyGraphicsView.swift

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct IdentifyGraphicsView: View {
+    /// A map with a topographic basemap.
+    @State private var map = Map(basemapStyle: .arcGISTopographic)
+    
+    /// A graphics overlay containing a yellow polygon graphic.
+    @State private var graphicsOverlay = {
+        let graphicsOverlay = GraphicsOverlay()
+        
+        // Create a polygon from a list of points.
+        let polygon = Polygon(points: [
+            Point(x: -20e5, y: 20e5),
+            Point(x: 20e5, y: 20e5),
+            Point(x: 20e5, y: -20e5),
+            Point(x: -20e5, y: -20e5)
+        ])
+        
+        // Create a graphic of the polygon and add it to the overlay.
+        let polygonGraphic = Graphic(geometry: polygon)
+        graphicsOverlay.addGraphic(polygonGraphic)
+        
+        // Create a renderer using a simple fill symbol.
+        let polygonSymbol = SimpleFillSymbol(style: .solid, color: .yellow)
+        graphicsOverlay.renderer = SimpleRenderer(symbol: polygonSymbol)
+        
+        return graphicsOverlay
+    }()
+    
+    /// A Boolean value indicating whether the graphic alert is showing.
+    @State private var isShowingGraphicAlert = false
+    
+    /// The text shown in the graphic alert that shows the alert when changed.
+    @State private var graphicAlertText = "" {
+        didSet { isShowingGraphicAlert = graphicAlertText.isNotEmpty }
+    }
+    
+    /// A Boolean value indicating whether to show an error alert.
+    @State private var isShowingErrorAlert = false
+    
+    /// The error shown in the error alert.
+    @State private var error: Error? {
+        didSet { isShowingErrorAlert = error != nil }
+    }
+    
+    var body: some View {
```

Comment:
This looks good but I would change this to following for doing identify operation in a `task` rather than in a closure of single tap gesture. Also, content of the result alert.

```swift
{
    /// A map with a topographic basemap.
    @State private var map = Map(basemapStyle: .arcGISTopographic)
    
    /// A graphics overlay containing a yellow polygon graphic.
    @State private var graphicsOverlay = {
        let graphicsOverlay = GraphicsOverlay()
        
        // Create a polygon from a list of points.
        let polygon = Polygon(points: [
            Point(x: -20e5, y: 20e5),
            Point(x: 20e5, y: 20e5),
            Point(x: 20e5, y: -20e5),
            Point(x: -20e5, y: -20e5)
        ])
        
        // Create a graphic of the polygon and add it to the overlay.
        let polygonGraphic = Graphic(geometry: polygon)
        graphicsOverlay.addGraphic(polygonGraphic)
        
        // Create a renderer using a simple fill symbol.
        let polygonSymbol = SimpleFillSymbol(style: .solid, color: .yellow)
        graphicsOverlay.renderer = SimpleRenderer(symbol: polygonSymbol)
        
        return graphicsOverlay
    }()
    
    /// A Boolean value indicating whether the graphic alert is showing.
    @State private var isShowingIdentifyResultAlert = false
    
    /// The text shown in the graphic alert that shows the alert when changed.
    @State private var identifyResultMessage = "" {
        didSet { isShowingIdentifyResultAlert = identifyResultMessage.isNotEmpty }
    }
    
    /// A Boolean value indicating whether to show an error alert.
    @State private var isShowingErrorAlert = false
    
    /// The error shown in the error alert.
    @State private var error: Error? {
        didSet { isShowingErrorAlert = error != nil }
    }
    
    /// Screen point for the identify operation.
    @State private var identifyScreenPoint: CGPoint?
    
    var body: some View {
        MapViewReader { proxy in
            MapView(map: map, graphicsOverlays: [graphicsOverlay])
                .onSingleTapGesture { screenPoint, _ in
                    identifyScreenPoint = screenPoint
                }
                .task(id: identifyScreenPoint) {
                    // Identify the screen point on the graphics overlay using the proxy.
                    // Use `identifyGraphicsOverlays` instead if you need to identify on
                    // all the graphics overlay present in the map view.
                    guard let identifyScreenPoint = identifyScreenPoint,
                          let identifyResult = await Result(awaiting: {
                              try await proxy.identify(
                                on: graphicsOverlay,
                                screenPoint: identifyScreenPoint,
                                tolerance: 12
                              )
                          }).cancellationToNil()
                    else {
                        return
                    }
                    
                    do {
                        // Display an alert if a graphic was found at the screen point.
                        let graphics = try identifyResult.get().graphics
                        if graphics.isNotEmpty {
                            self.identifyResultMessage = "\(graphics.count) graphic\(graphics.count > 1 ? "s" : "") found."
                        }
                    } catch {
                        // Show an error alert for an error throw while identifying.
                        self.error = error
                    }
                    
                    self.identifyScreenPoint = nil
                }
                .alert(
                    "Identify Result",
                    isPresented: $isShowingIdentifyResultAlert,
                    presenting: identifyResultMessage,
                    actions: { _ in },
                    message: { message in
                        Text(message)
                    }
                )
                .alert(
                    isPresented: $isShowingErrorAlert,
                    presentingError: error
                )
        }
    }
}
```

Reply:
I moved the identify operation to a `task` modifier, as you suggested, but made made it a little more readable since this is supposed to be a pretty simple sample. I also updated the alert but kept the message the same. Even though yours makes more sense, I think it would be better to stick to the design here since all the other sample implementations do it similarly. Let me know what you think.

Reply:
Ok